Formula/libnghttp2.rb

```diff
@@ -0,0 +1,42 @@
+class Libnghttp2 < Formula
```

Comment:
How much does building only the libraries speed up build time? I worry this adds some extra maintenance work without really all that much benefit.

Reply:
In terms of the actual build of `nghttp2` itself: probably not much.

This is purely with the idea that we can kill 3 dependencies from the `curl` dep tree so it might speed up that because you have to install less. And there was a couple of `nghttp2` dependencies which I could not find plain HTTP mirrors for.

One benefit this might have is relocatable bottles, which will definitely improve install time. (Double check CI output in case I'm wrong.)

Reply:
Yeah it is:

```
==> Bottling libnghttp2--1.45.1.arm64_big_sur.bottle.tar.gz...
/opt/homebrew/opt/gnu-tar/bin/gtar --create --numeric-owner --format pax --owner 0 --group 0 --sort name --pax-option globexthdr.name=/GlobalHead.%n,exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime --file /Users/brew/actions-runner-x86_64/_work/homebrew-core/homebrew-core/bottles/libnghttp2--1.45.1.arm64_big_sur.bottle.tar libnghttp2/1.45.1
./libnghttp2--1.45.1.arm64_big_sur.bottle.tar.gz
  bottle do
    sha256 cellar: :any, arm64_big_sur: "b6a81d7be284f3619e6d36551143152dc31a69e01e060e659397f5228f71fe89"
  end
Writing libnghttp2--1.45.1.arm64_big_sur.bottle.json
```

```
==> Bottling nghttp2--1.45.1_1.arm64_big_sur.bottle.tar.gz...
/opt/homebrew/opt/gnu-tar/bin/gtar --create --numeric-owner --format pax --owner 0 --group 0 --sort name --pax-option globexthdr.name=/GlobalHead.%n,exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime --file /Users/brew/actions-runner-x86_64/_work/homebrew-core/homebrew-core/bottles/nghttp2--1.45.1_1.arm64_big_sur.bottle.tar nghttp2/1.45.1_1
/opt/homebrew/Cellar/nghttp2/1.45.1_1/bin/nghttpx
 --> match '/opt/homebrew/Cellar/nghttp2/1.45.1_1/share/nghttp2/fetch-ocsp-response' at offset 0x8d601
Warning: String '/opt/homebrew/Cellar' still exists in these files:

./nghttp2--1.45.1_1.arm64_big_sur.bottle.tar.gz
  bottle do
    sha256 arm64_big_sur: "c3465c3d63b3dcd3c826db8fad58c6eb9c75dbb5e0f4c82ea6196d36f7217685"
  end
Writing nghttp2--1.45.1_1.arm64_big_sur.bottle.json
```